addons/website_sale/static/src/xml/website_sale_recently_viewed.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <!-- Recently Viewed Products Carousel -->
+    <t t-name="website_sale.productsRecentlyViewed">
+        <div t-att-id="uniqueId" class="carousel slide">
+            <div class="carousel-inner">
+            <t t-foreach="products.length" t-as="_index">
+                <t t-if="_index % productFrame === 0">
+                    <t t-set="_toIndex" t-value="_index + productFrame > products.length ? products.length : _index + productFrame"/>
```

Comment:
`Math.min(_index + productFrame, products.length)`

But in fact:
Damn it took me a while to figure out this rendering. Can't you just do:
```xml
<t t-set="productsGroups" t-value="_.chunk(products, productFrame)"/>
```
then iterate on those? Or even do not pass productFrame in js but directly the productsGroups this way?